src/Microsoft.Framework.Configuration.Abstractions/IConfigurationBuilder.cs

```diff
@@ -7,7 +7,7 @@ namespace Microsoft.Framework.Configuration
 {
     public interface IConfigurationBuilder
     {
-        string BasePath { get; }
+        Dictionary<string, object> Properties { get; }
```

Comment:
@Eilon does the name `Properties` seem ok for you for the property bag on `ConfigurationBuilder`? Other alternatives could be:

`ExtendedProperties`
`BuildTimeProperties` :trollface: